pkg/dom0-ztools/rootfs/etc/init.d/009-id

```diff
@@ -5,3 +5,16 @@
 ID=$(zboot curpart 2>/dev/null)
 echo "${ID:-RAM}" > /run/eve.id
 cp -p /etc/eve-release /run/eve-release
+v=$(cat /run/eve-release)
+echo $v | grep kubevirt
```

Comment:
We need more specific patterns to avoid having e.g., a branch name like erik-fix-kubevirt match.
In grub rootfs.cfg we use patterns like
      if regexp -- '-xen(-amd64$|-arm64$|-riscv64$|$)' $rootfs_title ; then
         eve_flavor=xen
      else
         eve_flavor=kvm
to avoid such false matches.

Here we should be able to extract the part of the string before the ISA to make it more general.

Reply:
I am surprised to know branch name gets added to eve image name. I thought its only commit hash and HV type.

Reply:
Why do string extraction? Why don't we just have an explicit entry in `eve-release` that says, "hv=<whatever>"?

Reply:
eve-release already exists, I am just using it. Its just the image name which already has HV included in it.

Reply:
Yeah, I get that. Still, teasing it out from an image name is less than reliable. It isn't your change, it is the general approach that predates it. How hard would it be to just include it in eve-release and be done?

Reply:
Nothing is hard, its just a software :). But don't know if it breaks any other assumptions in code or testing scripts. If someone can confirm it does not break anything, I can just add to eve release.

Reply:
We could switch to using hv=kvm, but wouldn't that imply that we should also change the ISA (amd64 etc) to isa=amd64?

And with such a syntax would we still want dashes separating the parts? Or commas?

I think the minimal change is to make this as robust as the current pattern matching. For other changes we should discuss the overall approach in the release naming scheme.

Reply:
IMO you don't need a new command here only to grab the hypervisor inside the version string. Some regex like the following can easily do the job: `eve version | sed "s/.*-\(kvm\|xen\|acrn\)-\(amd64\|arm64\|riscv64\).*/\1/"` without the needs of ifs/elses. Also, why ignore acrn? if an acrn image can be built, then it should be considered here as well...

Reply:
Actually I am thinking simplest change is to ship /etc/eve-hv-type and copy it  /run like we are doing with /etc/eve-release. Then we need not do anything and it's fixed at build time. I will update the commit if there is no issue with this approach.

Reply:
@zedi-pramodh are you then proposing to change all of the code (from grub and onwards) which today determines the eve-flavor/hv from the eve-release string so that there is no remaining read of that file (except for the check in https://github.com/lf-edge/eve/blob/fa6d8bd67e6caa274daf2c54e6c24cbcdaf351f6/pkg/pillar/zboot/zboot.go#L465)

Or would you like to add a new file and logic without removing the old usage?

Reply:
And the file content is generated from HV value used during make command.

Reply:
Updated PR, looks much cleaner now.